### ISSUE TITLE
Missing block proposal due to timeout

### ISSUE THREAD

**Issue by NeverAnIsland** [Consensys-Incorporated/teku#7063]
### Description
I had a missing proposal tonight. I use Teku + Nethermind (as part of RocketPool setup) and overall everything works fine. My validator client is located on another host than BN and EC.

### Versions
* Software version: consensys/teku:23.3.1 Docker container
* OS Name & Version: Ubuntu 20.04.6 LTS
* Docker Version: 23.0.2

### Context
<details>
<summary>Here is a part of BN log with the error:</summary>

```
2023-04-20 01:43:15.181 INFO  - Slot Event  *** Slot: 6260914, Block: 7ad3b8a5ec1aa4d903349fe323e9f24b708846754b3c12e29e4d431f77db6a98, Justified: 195
652, Finalized: 195651, Peers: 48                                                                                                                     
2023-04-20 01:43:27.214 INFO  - Slot Event  *** Slot: 6260915, Block:                                                        ... empty, Justified: 195
652, Finalized: 195651, Peers: 48                                                                                                                     
2023-04-20 01:43:27.624 WARN  - Late Block Import *** Block: 7791cc3d197c81c65d58f6956714f52a434cc9a2c3e0388c02fecfe8f2465314 (6260915) proposer 51742
9 arrival 3791ms, pre-state_retrieved +32ms, processed +67ms, execution_payload_result_received +733ms, begin_importing +0ms, transaction_prepared +0m
s, transaction_committed +0ms, completed +0ms                                                                                                         
2023-04-20 01:43:39.152 INFO  - Slot Event  *** Slot: 6260916, Block: f56a31d90238092aca97dae0331bf3df5f1f0effeb36b00c75a4062d7bc52f70, Justified: 195
652, Finalized: 195651, Peers: 48                                                                                                                     
2023-04-20 01:43:51.232 INFO  - Slot Event  *** Slot: 6260917, Block: 6be67effb76c91c43a2747f09af8f78b8c5fa5434a1023358a5b2613f2dc9593, Justified: 195
652, Finalized: 195651, Peers: 48                                                                                                                     
2023-04-20 01:44:03.001 INFO  - Slot Event  *** Slot: 6260918, Block: 5b9dad5cf9cf5220bb283324bccb995283b9824a8e6c8fa8157c321f2c99771f, Justified: 195
652, Finalized: 195651, Peers: 49                                                                                                                     
2023-04-20 01:44:15.174 INFO  - Slot Event  *** Slot: 6260919, Block: 9f389e4d7a4de65ec7d0bf7e5bd05ea13cd7a00d8402fd288083ab9f165f502a, Justified: 195
652, Finalized: 195651, Peers: 49
2023-04-20 01:44:24.385 INFO  - Received Builder Bid (Block Number = 17084562, Block Hash = 0xfbd5062d103a68e7421687b0b12c769849c3b39aa8e1afc6258be856
db112a78, MEV Reward (wei) = 89286311556919394, Gas Limit = 30000000, Gas Used = 17962195)
2023-04-20 01:44:27.097 INFO  - Slot Event  *** Slot: 6260920, Block:                                                        ... empty, Justified: 195
652, Finalized: 195651, Peers: 48
2023-04-20 01:44:39.251 INFO  - Slot Event  *** Slot: 6260921, Block: 16c040478e5a0c1fc15669c6475e8d7cb11eda2390ec59d4496637aec5ddea17, Justified: 195
652, Finalized: 195651, Peers: 47
2023-04-20 01:44:51.230 INFO  - Slot Event  *** Slot: 6260922, Block: 38b1e2e0e945f15d74aeaf1b700a10098660ea1ee83e603603b3b9a72e86f1b8, Justified: 195
652, Finalized: 195651, Peers: 47
2023-04-20 01:44:54.883 ERROR - Failed to process request to URL http://10.0.1.2:5052/eth/v1/beacon/blinded_blocks
java.io.UncheckedIOException: java.io.IOException: java.util.concurrent.TimeoutException: Idle timeout expired: 30000/30000 ms
at tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata.getRequestBody(EndpointMetadata.java:155) ~[teku-infrastructure-restapi-23.3.1.
jar:23.3.1]
at tech.pegasys.teku.infrastructure.restapi.endpoints.JavalinRestApiRequest.getRequestBody(JavalinRestApiRequest.java:51) ~[teku-infrastructure-restap
i-23.3.1.jar:23.3.1]
at tech.pegasys.teku.beaconrestapi.handlers.v1.beacon.PostBlindedBlock.handleRequest(PostBlindedBlock.java:76) ~[teku-data-beaconrestapi-23.3.1.jar:23
.3.1]
at tech.pegasys.teku.infrastructure.restapi.endpoints.JavalinEndpointAdapter.handle(JavalinEndpointAdapter.java:36) ~[teku-infrastructure-restapi-23.3
.1.jar:23.3.1]
at io.javalin.routing.HandlerEntry.handle(HandlerEntry.kt:19) ~[javalin-5.4.2.jar:5.4.2]
at io.javalin.http.servlet.DefaultTasks.HTTP$lambda-8$lambda-6$lambda-5(DefaultTasks.kt:38) ~[javalin-5.4.2.jar:5.4.2]
at io.javalin.http.servlet.JavalinServlet.handleTask(JavalinServlet.kt:88) ~[javalin-5.4.2.jar:5.4.2]
at io.javalin.http.servlet.JavalinServlet.handleSync(JavalinServlet.kt:53) ~[javalin-5.4.2.jar:5.4.2]
at io.javalin.http.servlet.JavalinServlet.service(JavalinServlet.kt:41) ~[javalin-5.4.2.jar:5.4.2]
at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
at io.javalin.jetty.JavalinJettyServlet.service(JavalinJettyServlet.kt:58) ~[javalin-5.4.2.jar:5.4.2]
at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[jetty-servlet-11.0.14.jar:11.0.14]
at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:529) ~[jetty-servlet-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[jetty-server-11.0.14.jar:11.0.14]
at io.javalin.jetty.JettyServer$start$wsAndHttpHandler$1.doHandle(JettyServer.kt:57) ~[javalin-5.4.2.jar:5.4.2]
at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484) ~[jetty-servlet-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1304) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.Server.handle(Server.java:563) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762) [jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497) [jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpChannel.run(HttpChannel.java:457) [jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:936) [jetty-util-11.0.14.jar:11.0.14]
at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1080) [jetty-util-11.0.14.jar:11.0.14]
at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.io.IOException: java.util.concurrent.TimeoutException: Idle timeout expired: 30000/30000 ms
at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:307) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:272) ~[jetty-server-11.0.14.jar:11.0.14]
at java.io.InputStream.readNBytes(Unknown Source) ~[?:?]
at java.io.InputStream.readAllBytes(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.restapi.openapi.request.OctetStreamRequestContentTypeDefinition.lambda$parseBytes$0(OctetStreamRequestContentTypeDefinition.java:35) ~[teku-infrastructure-restapi-23.3.1.jar:23.3.1]
at tech.pegasys.teku.infrastructure.restapi.openapi.request.OctetStreamRequestContentTypeDefinition.deserialize(OctetStreamRequestContentTypeDefinition.java:40) ~[teku-infrastructure-restapi-23.3.1.jar:23.3.1]
at tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata.getRequestBody(EndpointMetadata.java:147) ~[teku-infrastructure-restapi-23.3.1.jar:23.3.1]
... 32 more
Caused by: java.util.concurrent.TimeoutException: Idle timeout expired: 30000/30000 ms
at org.eclipse.jetty.io.IdleTimeout.checkIdleTimeout(IdleTimeout.java:170) ~[jetty-io-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.IdleTimeout.idleCheck(IdleTimeout.java:112) ~[jetty-io-11.0.14.jar:11.0.14]
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
... 1 more
Suppressed: java.lang.Throwable: HttpInput idle timeout
at org.eclipse.jetty.server.HttpChannelOverHttp.doOnIdleTimeout(HttpChannelOverHttp.java:697) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpChannelOverHttp.onIdleTimeout(HttpChannelOverHttp.java:684) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpConnection.onReadTimeout(HttpConnection.java:527) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.AbstractConnection.onFillInterestedFailed(AbstractConnection.java:168) ~[jetty-io-11.0.14.jar:11.0.14]
at org.eclipse.jetty.server.HttpConnection.onFillInterestedFailed(HttpConnection.java:534) ~[jetty-server-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.AbstractConnection$ReadCallback.failed(AbstractConnection.java:320) ~[jetty-io-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.FillInterest.onFail(FillInterest.java:135) ~[jetty-io-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.AbstractEndPoint.onIdleExpired(AbstractEndPoint.java:412) ~[jetty-io-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.IdleTimeout.checkIdleTimeout(IdleTimeout.java:170) ~[jetty-io-11.0.14.jar:11.0.14]
at org.eclipse.jetty.io.IdleTimeout.idleCheck(IdleTimeout.java:112) ~[jetty-io-11.0.14.jar:11.0.14]
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
at java.lang.Thread.run(Unknown Source) [?:?]
2023-04-20 01:45:03.154 INFO  - Slot Event  *** Slot: 6260923, Block: 2e88db462e65389c3d91c169d862be5fcaa030d84a4643d06d86d92cc47d01e7, Justified: 195
652, Finalized: 195651, Peers: 47
2023-04-20 01:45:15.244 INFO  - Slot Event  *** Slot: 6260924, Block: f30606e47ad40f07482d1911404a557ff116a3b0b2156291c01a36de31f32cdc, Justified: 195
652, Finalized: 195651, Peers: 47
2023-04-20 01:45:27.149 INFO  - Slot Event  *** Slot: 6260925, Block: 4c012c6f6fd6245f68d0bc895876d34be33eaf2533b62d1bb9049dd39db0a5c3, Justified: 195
652, Finalized: 195651, Peers: 47
2023-04-20 01:45:39.173 INFO  - Slot Event  *** Slot: 6260926, Block: bec131b52bc35c0e8e600f6dfd5142ee616ac297e0d1783beb603f2e9dafaa25, Justified: 195
652, Finalized: 195651, Peers: 47
```
</details>

<details>
<summary>Here is a part of VC log with the error:</summary>

```
2023-04-20 01:39:59.176 INFO  - Validator   *** 1 out of 1 validator registration(s) were successfully sent to the builder network via the Beacon Node.                                                                                                                                                     
2023-04-20 01:41:13.792 INFO  - Validator   *** Published attestation        Count: 1, Slot: 6260904, Root: 49406609843bb50e4ea042343d5f901cb81f52602dce583297119139c132f581                                                                                                                                
2023-04-20 01:44:54.947 ERROR - Validator   *** Failed to produce block  Slot: 6260920 Validator: b10f12a                                             tech.pegasys.teku.api.exceptions.RemoteServiceNotAvailableException: Server error from Beacon Node API (url = http://10.0.1.2:5052/eth/v1/beacon/blind
ed_blocks, status = 500, message = An unexpected error occurred)                                                                                      
at tech.pegasys.teku.validator.remote.typedef.ResponseHandler.serviceErrorHandler(ResponseHandler.java:117) ~[teku-validator-remote-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.ResponseHandler.handleResponse(ResponseHandler.java:93) ~[teku-validator-remote-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.handlers.AbstractTypeDefRequest.executeCall(AbstractTypeDefRequest.java:149) ~[teku-validator-remote-23.
3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.handlers.AbstractTypeDefRequest.postOctetStream(AbstractTypeDefRequest.java:142) ~[teku-validator-remote
-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.handlers.SendSignedBlockRequest.sendSignedBlockAsSsz(SendSignedBlockRequest.java:70) ~[teku-validator-re
mote-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.handlers.SendSignedBlockRequest.sendSignedBlockAsSszOrFallback(SendSignedBlockRequest.java:61) ~[teku-va
lidator-remote-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.handlers.SendSignedBlockRequest.sendSignedBlock(SendSignedBlockRequest.java:55) ~[teku-validator-remote-
23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.typedef.OkHttpValidatorTypeDefClient.sendSignedBlock(OkHttpValidatorTypeDefClient.java:87) ~[teku-validator-remo
te-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.lambda$sendSignedBlock$13(RemoteValidatorApiHandler.java:282) ~[teku-validator-remote-
23.3.1.jar:23.3.1]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:82) ~[teku-infrastructure-async-23.3.1.jar:23.3.1]
at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.sendRequest(RemoteValidatorApiHandler.java:483) ~[teku-validator-remote-23.3.1.jar:23.
3.1]
at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.lambda$sendRequest$34(RemoteValidatorApiHandler.java:478) ~[teku-validator-remote-23.3
.1.jar:23.3.1]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:74) ~[teku-infrastructure-async-23.3.1.jar:23.3.1]
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:124) ~[teku-
infrastructure-async-23.3.1.jar:23.3.1]
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
at java.lang.Thread.run(Unknown Source) [?:?]
2023-04-20 01:46:23.144 INFO  - Validator   *** 1 out of 1 validator registration(s) were successfully sent to the builder network via the Beacon Node
.
2023-04-20 01:48:37.623 INFO  - Validator   *** Published attestation        Count: 1, Slot: 6260941, Root: 596b5be0d561eeca011a22b0a20f611e7c3afd955a
74e51411bbe5b5975c7695
```
</details>

<details>
<summary>Here is a part of EC log with the error:</summary>

```
2023-04-20 01:44:13.3287|Received a new payload: 17084561 (0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f) 
2023-04-20 01:44:14.0065|Processed   17084561 |      661ms of  13,514ms, mgasps   36.29 total   34.36, tps  420.72 total  334.29, bps    1.51 total   
 1.13, recv queue 0, proc queue 0 
2023-04-20 01:44:14.0065|Valid. Result of a new payload: 17084561 (0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f). 
2023-04-20 01:44:14.2984|Received: ForkchoiceState: (HeadBlockHash: 0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f, SafeBlockHash:
 0x645a7091f19cfd84fd922a666efcab77c298e328f778eb6f1124b7fe240a0cd9, FinalizedBlockHash: 0x2e4210f521f8c6c15e6d170194b5d79e8738136bbbb422c8b5709325e5c
8ea6e) PayloadAttributes {Timestamp: 1681955063, PrevRandao: 0x3de887347d93cbaca28cc2d6bb9cfa197b2bb1c1aa00191ae57493307f3d7c65, SuggestedFeeRecipient
: 0xd4e96ef8eee8678dbff4d535e033ed1a4f7605b7, Withdrawals count: 16} 
2023-04-20 01:44:14.2984|FCU - block 17084561 (0xc0ad4e...78c05f) was processed. 
2023-04-20 01:44:14.2984|Block 0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f was set as head. 
2023-04-20 01:44:14.8503|Sealed block 17084562 (0x1702e0...76c7ab), diff: 0, tx count: 58 
2023-04-20 01:44:14.8503|Valid. Request: ForkchoiceState: (HeadBlockHash: 0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f, SafeBloc
kHash: 0x645a7091f19cfd84fd922a666efcab77c298e328f778eb6f1124b7fe240a0cd9, FinalizedBlockHash: 0x2e4210f521f8c6c15e6d170194b5d79e8738136bbbb422c8b5709
325e5c8ea6e) PayloadAttributes {Timestamp: 1681955063, PrevRandao: 0x3de887347d93cbaca28cc2d6bb9cfa197b2bb1c1aa00191ae57493307f3d7c65, SuggestedFeeRec
ipient: 0xd4e96ef8eee8678dbff4d535e033ed1a4f7605b7, Withdrawals count: 16}. 
2023-04-20 01:44:14.8579|Improved post-merge block 17084562 (0x1702e0...76c7ab), diff: 0, tx count: 58 
2023-04-20 01:44:18.0189|Sealed block 17084562 (0x016672...b56877), diff: 0, tx count: 100 
2023-04-20 01:44:18.0189|Improved post-merge block 17084562 (0x016672...b56877), diff: 0, tx count: 100 
2023-04-20 01:44:21.1589|Sealed block 17084562 (0x123f41...ed02aa), diff: 0, tx count: 131 
2023-04-20 01:44:21.1589|Improved post-merge block 17084562 (0x123f41...ed02aa), diff: 0, tx count: 131 
2023-04-20 01:44:21.1938|Peers | with known best block: 50 | all: 50 | 
2023-04-20 01:44:23.4462|GetPayloadV2 result: Hash: 0x123f41c34d64193da60e428db72bd107fe82a4a50f0de2f86e4bd6dd4eed02aa
Number: 17084562
Parent: 0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f
Beneficiary: 0xd4e96ef8eee8678dbff4d535e033ed1a4f7605b7
Gas Limit: 30000000
Gas Used: 12538672
Timestamp: 1681955063
Extra Data: 4e65746865726d696e64
Difficulty: 0
Mix Hash: 0x3de887347d93cbaca28cc2d6bb9cfa197b2bb1c1aa00191ae57493307f3d7c65
Nonce: 0
Uncles Hash: 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347
Tx Root: 0xe87624f87b5c7eda3656627b586180a14e540d238dea27b8aa650fa6869f237e
Receipts Root: 0xf267520353eda02f589e3a4460c9dfd475f540e5044459084b4c183481d973c6
State Root: 0xb545f365fd15e23949c350780c0133f7e9fbaa1fbdb78d54980f0046d18a85fe
BaseFeePerGas: 56996634685
WithdrawalsRoot: 0xfc4a7a75a7bb8d18eca150682c581f90f192aef7452286e062ba8b4c3a94cfd1
IsPostMerge: True
TotalDifficulty: 58750003716598352816469
. 
2023-04-20 01:44:27.1007|Received: ForkchoiceState: (HeadBlockHash: 0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f, SafeBlockHash: 0x645a7091f19cfd84fd922a666efcab77c298e328f778eb6f1124b7fe240a0cd9, FinalizedBlockHash: 0x2e4210f521f8c6c15e6d170194b5d79e8738136bbbb422c8b5709325e5c8ea6e)  
2023-04-20 01:44:27.1007|FCU - block 17084561 (0xc0ad4e...78c05f) was processed. 
2023-04-20 01:44:27.1007|Valid. Request: ForkchoiceState: (HeadBlockHash: 0xc0ad4e398b39d52b2103f3837b76ec8ff25640e70c92c96cd7003aa63078c05f, SafeBlockHash: 0x645a7091f19cfd84fd922a666efcab77c298e328f778eb6f1124b7fe240a0cd9, FinalizedBlockHash: 0x2e4210f521f8c6c15e6d170194b5d79e8738136bbbb422c8b5709325e5c8ea6e) . 
2023-04-20 01:44:36.1955|Peers | with known best block: 48 | all: 48 | 
2023-04-20 01:44:36.9542|Received a new payload: 17084562 (0x0e541964f9cf64be7a1800582fe865d931937c55fc87aa46a532c1ebf985fc6a) 
2023-04-20 01:44:37.4733|Processed   17084562 |      507ms of  23,467ms, mgasps   69.88 total   34.36, tps  763.03 total  334.29, bps    1.97 total    1.13, recv queue 0, proc queue 0 
2023-04-20 01:44:37.4733|Valid. Result of a new payload: 17084562 (0x0e541964f9cf64be7a1800582fe865d931937c55fc87aa46a532c1ebf985fc6a).
```
</details>

**Comment by rolfyone** [comment]
Thanks for raising..

I can see the BN log timing out after 30 seconds, it's basically waiting on the VC, and the VC stops because the BN terminates the long running operation...

What I'm not sure of is why the VC would not complete, as it won't be a large amount of data that it's sending.

Is there an external signer involved? 
Is the VC the same version as the BN? 
Is there anything more in the earlier logs from the VC? Logs from around `2023-04-20 01:44:20` - `2023-04-20 01:44:30`

I might loop in some more people to get more information also, but these would be my initial questions...

**Comment by NeverAnIsland** [comment]
There's no external signer.
VC and BN are the same version.
There is nothing suspicious in VC logs before or after this, just messages about published attestations and validator registrations.

**Comment by tbenr** [comment]
I agree with @rolfyone that from logs it seems BN was waiting VC data stream (signed blinded block) to complete and timedout after 30 seconds.
Are VC and BN on the same physical machine?

**Comment by rolfyone** [comment]
Has this happened since the initial incident @NeverAnIsland?

**Comment by NeverAnIsland** [comment]
Sorry for a late reply, I was on a trip away from my laptop. I have EC an BN on one physical machine and VC on a cloud instance. I've got but one validator and wasn't that lucky to have one more proposal yet.

**Comment by tbenr** [comment]
So if I understand correctly, VC interacts with BN apis via Internet (maybe via a VPN?), they are not on the same local network. This explains the error seen on the BN side, which seems to show HTTP POST of the signed blinded block from VC has been terminated by the BN after 30 seconds of idle time. I'd check your networking infrastructure, there might have been some outages or difficulties handling that POST.

**Comment by NeverAnIsland** [comment]
Yes there is WireGuard tunnel between parties. However I run this setup for many months and it worked well for both attestations and proposals. Does payload for a blinded block differ significantly from attestation's?

**Comment by rolfyone** [comment]
It's basically just a lot larger than attestations, and also there's interaction with the builder, where as attestations don't have that problem... blinded blocks should at least be smaller than full blocks...

You'd definitely want all your MEV components co-located, BN+EL+builder, and the risk becomes if that link is slow that you don't get data to/from fast enough to sign in a timely way. If your blocks is delayed by even a small amount you may miss the proposal, as there's many steps in the process, and builders tend to produce as late as possible to not incur penalties, which works against you if you're adding latency...

Having said that an actual 30 second timeout tends to suggest the link was maybe down at just the wrong moment...

Is the separation needed? It's a very time critical process and latency would not be your friend...

**Comment by rolfyone** [comment]
@NeverAnIsland we'll close this for now, there's not really much more we can advise without further information.

Please feel free to either re-open or open a new issue if something re-occurs or theres more information we can assist with.
